src/components/clusterConfiguration/NetworkConfigurationForm.tsx

```diff
@@ -0,0 +1,201 @@
+  */
+  const { onBack, onNext } = useClusterWizardTransitionFunctions(CLUSTER_STEP_NETWORKING);
+  const { alerts, addAlert, clearAlerts } = React.useContext(AlertsContext);
+  const dispatch = useDispatch();
+  const hostSubnets = React.useMemo(() => getHostSubnets(cluster), [cluster]);
+  const initialValues = React.useMemo(() => getNetworkInitialValues(cluster, managedDomains), [
+    cluster,
+    managedDomains,
+  ]);
+
+  const memoizedValidationSchema = React.useMemo(
+    () => validationSchema(initialValues, hostSubnets),
+    [hostSubnets, initialValues],
+  );
+
+  const handleSubmit = async (
+    values: NetworkConfigurationValues,
+    formikActions: FormikHelpers<NetworkConfigurationValues>,
+  ) => {
+    clearAlerts();
+
+    // update the cluster configuration
+    try {
+      const params = _.omit(values, ['hostSubnet', 'useRedHatDnsService', 'shareDiscoverySshKey']);
+
+      // Discard additionalNtpSource if it is empty
+      if (!values.additionalNtpSource) {
+        delete params.additionalNtpSource;
+      }
+
+      if (values.shareDiscoverySshKey) {
+        params.sshPublicKey = cluster.imageInfo.sshPublicKey;
+      }
+
+      if (values.vipDhcpAllocation) {
+        delete params.apiVip;
+        delete params.ingressVip;
+        const cidr = hostSubnets.find((hn) => hn.humanized === values.hostSubnet)?.subnet;
+        params.machineNetworkCidr = cidr;
+      }
+
+      const { data } = await patchCluster(cluster.id, params);
+      formikActions.resetForm({
+        values: getNetworkInitialValues(data, managedDomains),
+      });
+      dispatch(updateCluster(data));
+
+      if (canNextNetwork({ cluster })) {
+        onNext && onNext();
+      }
+    } catch (e) {
+      handleApiError<ClusterUpdateParams>(e, () =>
+        addAlert({ title: 'Failed to update the cluster', message: getErrorMessage(e) }),
+      );
+    }
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={memoizedValidationSchema}
+      onSubmit={handleSubmit}
+      initialTouched={_.mapValues(initialValues, () => true)}
+      validateOnMount
+    >
+      {({
+        isSubmitting,
+        isValid,
+        dirty,
+        values,
+        errors,
+        submitForm,
+        setFieldValue,
+      }: FormikProps<NetworkConfigurationValues>) => {
+        const onClusterSshKeyToggle = (isChecked: boolean) =>
+          setFieldValue('shareDiscoverySshKey', isChecked);
+        const onClusterSshKeyVisibilityChanged = () => {
+          onClusterSshKeyToggle(
+            !!cluster.imageInfo.sshPublicKey &&
+              (cluster.sshPublicKey === cluster.imageInfo.sshPublicKey || !cluster.sshPublicKey),
+          );
+        };
+        const onSshKeyBlur = () => {
+          if (values.sshPublicKey) {
+            setFieldValue('sshPublicKey', trimSshPublicKey(values.sshPublicKey));
+          }
+        };
+
+        const form = (
+          <Form>
```

Comment:
Wrap the `Form` with a grid to limit fields width
```suggestion
          <Grid hasGutter>
            <GridItem span={12} lg={10} xl={9} xl2={7}>
              <Form>
```